skyportal/handlers/thumbnail.py

```
@@ -113,22 +115,23 @@ def create_thumbnail(thumbnail_data, thumbnail_type, source_id, photometry_obj):
     file_bytes = base64.b64decode(thumbnail_data)
     im = Image.open(io.BytesIO(file_bytes))
     if im.format != 'PNG':
-        return self.error('Invalid thumbnail image type. Only PNG are supported.')
+        raise TypeError('Invalid thumbnail image type. Only PNG are supported.')
     if not (100, 100) <= im.size <= (500, 500):
-        return self.error('Invalid thumbnail size. Only thumbnails '
-                          'between (100, 100) and (500, 500) allowed.')
+        raise TypeError('Invalid thumbnail size. Only thumbnails '
+                        'between (100, 100) and (500, 500) allowed.')
     try:
         t = Thumbnail(type=thumbnail_type,
                       photometry=photometry_obj,
                       file_uri=file_uri,
                       public_url=f'/static/thumbnails/{source_id}_{thumbnail_type}.png')
         DBSession.add(t)
-    except TypeError:
-        return self.error('Invalid thumbnail type. Please refer to '
+        DBSession.flush()
+    except LookupError:
+        raise LookupError('Invalid thumbnail type. Please refer to '
                           'API docs for a list of allowed types.')
     except Exception as e:
-        return self.error(f'Error creating thumbnail: {str(e)}. Please check '
-                          'submitted values against API docs.')
+        raise Exception(f'Error creating thumbnail: {str(e)}. Please check '
```

Perhaps better to just re-raise this exception.